src/libraries/ExcelConverter/ExcelParser.cs

```diff
@@ -66,352 +66,370 @@ public sealed class ParsedCell
 
-                using (SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Open(excelFilePath, false))
-                {
-                    var sheets = new Dictionary<string, Sheet>(spreadsheetDocument.WorkbookPart.Workbook.Sheets.Elements<Sheet>().Select(s => new KeyValuePair<string, Sheet>(s.Id.Value, s)));
+            return result;
+        }
 
-                    var jsonRows = new List<ParsedCell>();
-                    List<ParsedTable> jsonTables = new List<ParsedTable>();
+        public static ParsedExcelData ParseSpreadsheet(String excelFilePath, bool outputFile = true)
+        {
+            string fileName = new FileInfo(excelFilePath).Name;
+            ParsedExcelData result;
+            using (SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Open(excelFilePath, false))
+            {
+                result = ParseSpreadSheetInternal(spreadsheetDocument, outputFile, fileName);
+            }
 
-                    foreach (var workSheetPartInfo in spreadsheetDocument.WorkbookPart.Parts.Where(p => p.OpenXmlPart is WorksheetPart))
-                    {
-                        Sheet sheet = sheets[workSheetPartInfo.RelationshipId];
-                        Worksheet workSheet = ((WorksheetPart)workSheetPartInfo.OpenXmlPart).Worksheet;
+            return result;
+        }
 
-                        Console.WriteLine(sheet.Name);
+        private static ParsedExcelData ParseSpreadSheetInternal(SpreadsheetDocument spreadsheetDocument, bool outputFile = true, string outputFileName = "")
+        {
+            bool includeCellValues = true;
 
-                        SheetData sheetData = workSheet.Elements<SheetData>().First();
-                        SharedStringItem[] sharedStringTable = spreadsheetDocument.WorkbookPart.SharedStringTablePart?.SharedStringTable.Elements<SharedStringItem>().ToArray();    
+            var sheets = new Dictionary<string, Sheet>(spreadsheetDocument.WorkbookPart.Workbook.Sheets.Elements<Sheet>().Select(s => new KeyValuePair<string, Sheet>(s.Id.Value, s)));
 
-                        var sharedFormulas = new Dictionary<uint, SharedFormula>();
+            var jsonRows = new List<ParsedCell>();
+            List<ParsedTable> jsonTables = new List<ParsedTable>();
 
+            foreach (var workSheetPartInfo in spreadsheetDocument.WorkbookPart.Parts.Where(p => p.OpenXmlPart is WorksheetPart))
+            {
+                Sheet sheet = sheets[workSheetPartInfo.RelationshipId];
+                Worksheet workSheet = ((WorksheetPart)workSheetPartInfo.OpenXmlPart).Worksheet;
 
-                        foreach (Row row in sheetData.Elements<Row>())
-                        {
-                            foreach (Cell cell in row.Elements<Cell>())
-                            {
-                                ParsedCell parsedCell = ParseCell(sheet.Name?.Value, cell, sharedStringTable, sharedFormulas, includeCellValues);
+                Console.WriteLine(sheet.Name);
```

Comment:
Should we be moving this to a logger class and having the console fill that need, or do we always anticipate the console as being the output?

Reply:
Right now this is just logging results to the console. I think we'll look and see how the core PFx is logging and adhere to that approach.